txpool/event_manager.go

```diff
@@ -0,0 +1,82 @@
+package txpool
+
+import (
+	"fmt"
+	"github.com/0xPolygon/polygon-sdk/txpool/proto"
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-hclog"
+	"sync"
+)
+
+type eventManager struct {
```

Comment:
No test for this? By experience, there are always bugs on the first iteration of an event manager. Also please consider using this https://github.com/hashicorp/go-eventlogger instead.

Reply:
Tests are still pending for this PR, along with the docs. I'll work on this and it should be wrapped up within a few days 👍 

Thank you for recommending the event logger 🙏 

It looks like a great opportunity to create a common event framework for not just the txpool, but for all modules. I'll add it as a task to our next sprint so we have a standardized way of handling events 💯